internal/configs/bootnode/bootnode.go

```diff
@@ -0,0 +1,397 @@
+package bootnode
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	nodeconfig "github.com/harmony-one/harmony/internal/configs/node"
+	"github.com/harmony-one/harmony/internal/utils"
+)
+
+// BootNodeConfig contains all the configs user can set for running harmony binary. Served as the bridge
+// from user set flags to internal node configs. Also user can persist this structure to a toml file
+// to avoid inputting all arguments.
+type BootNodeConfig struct {
+	Version    string
+	General    GeneralConfig
+	Network    NetworkConfig
+	P2P        P2pConfig
+	HTTP       HttpConfig
+	WS         WsConfig
+	RPCOpt     RpcOptConfig
+	BLSKeys    BlsConfig
+	Pprof      PprofConfig
+	Log        LogConfig
+	Sys        *SysConfig        `toml:",omitempty"`
+	Devnet     *DevnetConfig     `toml:",omitempty"`
+	Revert     *RevertConfig     `toml:",omitempty"`
```

Comment:
What's the use case for a revert function on the bootnode ?

Reply:
same question for BlsConfig, ShardDataConfig, CacheConfig, LegacyConfig, DevnetConfig, SysConfig 

in cmd/bootnode/main.go there is only a couple of new flag added.

Reply:
all are removed